docs/tutorials/extending_admin_ui/3b_adding_edit_button_content_list.md

```diff
@@ -0,0 +1,229 @@
+    use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+    use eZ\Publish\Core\Pagination\Pagerfanta\LocationSearchAdapter;
+    use Pagerfanta\Pagerfanta;
+    
+    class AllContentListController extends Controller
+    {
+        private $searchService;
+    
+        private $contentTypeService;
+    
+        public function __construct(SearchService $searchService, ContentTypeService $contentTypeService, FormFactory $formFactory)
+        {
+            $this->searchService = $searchService;
+            $this->contentTypeService = $contentTypeService;
+            $this->formFactory = $formFactory;
+        }
+    
+        public function listAction($page = 1)
+        {
+            $query = new LocationQuery();        $criterions = [
+            new Criterion\Visibility(Criterion\Visibility::VISIBLE),
+        ];        $query->query = new Criterion\LogicalAnd($criterions);        $paginator = new Pagerfanta(
+            new LocationSearchAdapter($query, $this->searchService)
+        );
+            $paginator->setMaxPerPage(8);
+            $paginator->setCurrentPage($page);
+            $editForm = $this->formFactory->contentEdit();
+            return $this->render('list/all_content_list.html.twig', [
+                'totalCount' => $paginator->getNbResults(),
+                'articles' => $paginator,
+                'form_edit' => $editForm->createView(),
+            ]);
+        }
+    }
+    ```
+
+## Change the template
+
+The last thing to do is to add the edit button to the content list template.
+All below code blocks should be added to `templates/list/all_content_list.html.twig`.
+
+First, add `<th>{{ 'Edit'|trans }}</th>` header to the content list table inside `<section class="container my-4">`:
+
+```html+twig  hl_lines="8"
+<table class="table">
+    <thead>
+    <tr>
+        <th>{{ 'Content name'|trans }}</th>
+        <th>{{ 'Content Type'|trans }}</th>
+        <th>{{ 'Modified'|trans }}</th>
+        <th>{{ 'Published'|trans }}</th>
+        <th>{{ 'Edit'|trans }}</th>
+    </tr>
+    </thead>
+```
+
+Next, add the edit button as a new <td> tag inside `<section class="container my-4">`:
+
+```html+twig
+<td>
+    <button class="btn btn-icon mx-2 ez-btn--content-edit"
+            title="{{ 'dashboard.table.all.content.edit'|trans|desc('Edit Content') }}"
+            data-content-id="{{ article.contentInfo.id }}"
+            data-version-no="{{ article.contentInfo.currentVersionNo }}"
+            data-language-code="{{ article.contentInfo.mainLanguageCode }}">
+        <svg class="ez-icon ez-icon-edit">
+            <use xlink:href="{{ asset('bundles/ezplatformadminui/img/ez-icons.svg') }}#edit"></use>
+        </svg>
+    </button>
+</td>
+```
+
+After that, add a hidden form with js listeners for draft conflict checks by adding the following code block inside `{%- block content -%}`, right under `<section class="container my-4">`:
```

Comment:
those are two different concepts, form is used for redirecting to content edit page as we dont use `get` request for that kind of users actions.

also this form does not have any js listeners, it is used by them.

listeners are in `{% block javascripts %}` block.